### PR TITLE
Adding more commands

### PR DESCRIPTION
**Pull request by rolandtritsch**
Hi there,

thanks for the `mermaid-mode` !!!

This PR ...
- adds a couple more commands
- and suggests to map to `C-c m ...` (because the `C-c` keyspace is kind-a crowded)
- and adds a screencast to the README
- and fixes a couple of minor whitespace/formatting issues

Feel free to edit and merge as you see fit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,27 +1,41 @@
 [![MELPA](https://melpa.org/packages/mermaid-mode-badge.svg)](https://melpa.org/#/mermaid-mode)
 
 # mermaid-mode
+
 Emacs major mode for working with [mermaid graphs](https://mermaidjs.github.io/)
 
-![alt text](screenshot.jpg "Screenshot")
+![Screenshot](./assets/screenshot.jpg "Screenshot")
+
+[![Thumbnail](./assets/thumbnail.png)](https://vimeo.com/414458581 "Screencast")
```
Comment (abrochard) [inline README.md:9]: Love it!

File: README.md [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
 
 ## Installation
+
 1. Install from Melpa or load the `mermaid-mode.el` file
-2. Install `mmdc` binary from the [mermaid.cli project](https://github.com/mermaidjs/mermaid.cli) if you plan to compile graphs in Emacs
+1. Install `mmdc` binary from the [mermaid.cli project](https://github.com/mermaidjs/mermaid.cli) if you plan to compile graphs in Emacs
```
Comment (abrochard) [inline README.md:14]: typo?

**Comment by abrochard** [comment]
Thank you for the PR! I am glad you like the extension.

The changes look good overall but I have reserves around the keybindings. The [convention](https://www.gnu.org/software/emacs/manual/html_node/elisp/Key-Binding-Conventions.html) is to give `C-c [key]` to minor modes and `C-c C-[key]` to major modes.

I also really like the `C-c C-c` to compile current file/buffer as it's pretty common for major modes (I think python-mode has the same binding?). I agree that `C-c C-d` for doc isn't great and even if it breaks backward compatibility, I'm ok to change it.

I also wonder if some of the interactive functions couldn't be consolidated? Maybe using implicit behavior or the `C-u` argument? Like for example, if the region is active `C-c C-c` compiles it, and if there is no region it applies to the buffer.

Ideally I think there should be as few bindings as necessary to the user to have to memorize. What kind of workflow do you use with it?

File: README.md (hunk 2 above)
Comment (rolandtritsch) [reply README.md:14]: No. I normally let the markdown do the numbering. Then I do not have to re-number everything when I add something in the middle of the list.

**Comment by rolandtritsch** [comment]
Good point ...

The reason I ended up with this key-binding is ...

* I want to use mermaid in markdown on github (which is the big selling point over `plantuml`)
* to edit the embedded markdown you probably want/need to use `edit-indirect`
* `edit-indirect` binds `C-c C-c` to `edit-indirect-commit` (to end the session/save the buffer) and I do not want to fiddle with that (because it would create an unexpected user experience)
* then I looked at `C-c C-m c` to find out that `C-m` is (obviously) bound to `RET`
* means my best suggestion was `C-c m c`
* but your point about the convention is very valid. Means we could/should also look at `C-c C-d c` (the `d` coming from `mermai(d)`)

What do you think?

**Comment by abrochard** [comment]
Hi @rolandtritsch ,
That makes sense but if that's your use-case, I suggest you directly edit the mermaid-mode-map in your config to what fits you best. For example:

```
(setq mermaid-mode-map
      (let ((map mermaid-mode-map))
        (define-key map (kbd "C-c C-c") nil) ;; unbind C-c C-c
        (define-key map (kbd "C-c m c") 'mermaid-compile) ;; rebind to C-c m c
        map))
```

Alternatively, I wonder if there's a way to detect that you are in an `edit-indirect` buffer and dynamically rebind.

**Comment by rolandtritsch** [comment]
Sorry. Got distracted.

Here we go.

Reverted to the old keymap and documented how to customize it.

Ready to merge?

**Comment by abrochard** [comment]
Thank you!

**Pull request merged**